internal/venice-common/src/main/java/com/linkedin/venice/pubsub/PubSubPositionFactory.java

```diff
@@ -0,0 +1,56 @@
+package com.linkedin.venice.pubsub;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.pubsub.adapter.kafka.ApacheKafkaOffsetPosition;
+import com.linkedin.venice.pubsub.api.PubSubPosition;
+import com.linkedin.venice.pubsub.api.PubSubPositionWireFormat;
+import com.linkedin.venice.serialization.avro.AvroProtocolDefinition;
+import com.linkedin.venice.serialization.avro.InternalAvroSpecificSerializer;
+import java.io.IOException;
+
+
+/**
+ * Factory class for creating PubSubPosition objects from wire format
+ */
+public class PubSubPositionFactory {
+  public static final PubSubPositionFactory INSTANCE = new PubSubPositionFactory();
+
+  /**
+   * Converts a wire format position to a PubSubPosition
+   * @param positionWireFormat the wire format position
+   * @return concrete position object represented by the wire format
+   */
+  PubSubPosition convertToPosition(PubSubPositionWireFormat positionWireFormat) {
+    if (positionWireFormat == null) {
+      throw new IllegalArgumentException("Cannot deserialize null wire format position");
+    }
+
+    if (positionWireFormat.type == PubSubPositionType.APACHE_KAFKA_OFFSET) {
```

Comment:
Nitpick: I assume when we add more open source pub sub implementations, this will eventually become a switch/case. Shall we make it a switch to begin with?

Reply:
Yeah, will replace `if` with `switch`.

Reply:
Done